railties/lib/rails/generators/rails/scaffold_controller/templates/controller.rb

```diff
@@ -64,5 +112,16 @@ def <%= "#{singular_table_name}_params" %>
       params.require(:<%= singular_table_name %>).permit(<%= attributes_names.map { |name| ":#{name}" }.join(', ') %>)
```

Comment:
I think you might have to change the line above (62/110) to `params.permit(:singular_table_name)`, so the generated method changes from:

```ruby
    # Only allow a trusted parameter "white list" through.    def post_params
      params[:post]
    end
```

to

```ruby
    # Only allow a trusted parameter "white list" through.
    def post_params
      params.permit(:post)
    end
```

If you don't do that and try to update a single post without changing any value, you get the following error:

```sh
Started PATCH "/posts/1" for ::1 at 2015-01-12 12:21:34 -0800
Processing by PostsController#update as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"fNc..ag==", "commit"=>"Update Post", "id"=>"1"}
D, [2015-01-12T12:21:34.302525 #19897] DEBUG -- :   Post Load (0.1ms)  SELECT  "posts".* FROM "posts" WHERE "posts"."id" = ? LIMIT 1  [["id", 1]]
D, [2015-01-12T12:21:34.302773 #19897] DEBUG -- :    (0.0ms)  begin transaction
D, [2015-01-12T12:21:34.302917 #19897] DEBUG -- :    (0.0ms)  rollback transaction
Completed 500 Internal Server Error in 1ms (ActiveRecord: 0.2ms)

ArgumentError (When assigning attributes, you must pass a hash as an argument.):
  app/controllers/posts_controller.rb:39:in `update'
```

whereas, if you use `params.permit(:post)`, then you get the expected result:

```sh
Started PATCH "/posts/1" for ::1 at 2015-01-12 12:15:10 -0800
Processing by PostsController#update as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"61f..==", "commit"=>"Update Post", "id"=>"1"}
D, [2015-01-12T12:15:10.200654 #19897] DEBUG -- :   Post Load (0.1ms)  SELECT  "posts".* FROM "posts" WHERE "posts"."id" = ? LIMIT 1  [["id", 1]]
Unpermitted parameters: utf8, _method, authenticity_token, commit, id
D, [2015-01-12T12:15:10.202163 #19897] DEBUG -- :    (0.0ms)  begin transaction
D, [2015-01-12T12:15:10.202734 #19897] DEBUG -- :    (0.0ms)  commit transaction
Redirected to http://localhost:3000/posts/1
Completed 302 Found in 5ms (ActiveRecord: 0.5ms)
```